src/View/SSTemplateParser.peg

```diff
@@ -1035,23 +1026,23 @@ class SSTemplateParser extends Parser implements TemplateParser
                 'arguments only.', $this);
         }
 
-        //loop without arguments loops on the current scope
+        // loop without arguments loops on the current scope
         if ($res['ArgumentCount'] == 0) {
-            $on = '$scope->locally()->obj(\'Me\', [], true)';
+            $on = '$scope->locally()->self()';
```

Comment:
I've made this change to allow for things like `ArrayIterator` to be used, which doesn't have a `$Me`.
I made this change before I decided to add `$Me` as a special name in `ViewLayerData`, but I still think it's a good change since it cuts out a bunch of unnecessary code calls to get the current scoped item.